### PR TITLE
Refactor special_euclidean.py for _plotter.py and corrected import in _plotter.py

### PR DESCRIPTION
**Pull request by SamFeinstein**
## Checklist

- [ x] My pull request has a clear and explanatory title.
- [x ] If necessary, my code is [vectorized](https://www.geeksforgeeks.org/vectorization-in-python/).
- [x ] I added appropriate unit tests.
- [x ] I made sure the code passes all unit tests. (refer to comment below) (there seemed to be other unrelated errors?)
- [x ] My PR follows [PEP8](https://peps.python.org/pep-0008/) guidelines. (refer to comment below)
- [x ] My PR follows [geomstats coding style](https://github.com/geomstats/geomstats/blob/master/docs/contributing.rst#coding-style-guidelines) and API.
- [x ] My code is properly documented and I made sure the documentation renders properly. ([Link](https://github.com/geomstats/geomstats/blob/master/docs/contributing.rst#documentation))
- [x ] I linked to issues and PRs that are relevant to this PR.

### CODE REVIEW

**Comment by SamFeinstein** [comment]
Still some errors from files that others are refactoring.

File: geomstats/visualization/_plotting.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
+import abc
```
Comment (ninamiolane) [inline geomstats/visualization/_plotting.py:1]: Need docstring a the top of this file, see other geomstats' files for inspiration

File: geomstats/visualization/_plotting.py [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+
+import matplotlib.pyplot as plt
+
+import geomstats._backend as gs
+
+
+class Plotter(metaclass=abc.ABCMeta):
```
Comment (ninamiolane) [inline geomstats/visualization/_plotting.py:8]: Need docstring for the class and each of its method

File: geomstats/visualization/_plotting.py [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+    def set_ax(self, ax=None):
+        """Set axis."""
+        if ax is not None:
+            return ax
+
+        ax = self._create_ax()
+        return self.config_ax(ax)
+
+    def _check_points(self, points):
+        # TODO: points may need to be converted to numpy
```
Comment (ninamiolane) [inline geomstats/visualization/_plotting.py:59]: Remove the two TODO lines

File: geomstats/visualization/hypersphere.py [hunk 4]
```diff
@@ -5,155 +5,100 @@
...
+from geomstats.visualization._plotting import Plotter
 
-S1 = Hypersphere(dim=1)
-S2 = Hypersphere(dim=2)
 
-AX_SCALE = 1.2
-
-
-class Circle:
+class Circle(Plotter):
```
Comment (ninamiolane) [inline geomstats/visualization/hypersphere.py:11]: Nice!

File: geomstats/visualization/hypersphere.py [hunk 5]
```diff
@@ -5,155 +5,100 @@
...
-        points = gs.array(points)
-        ax.plot(points[:, 0], points[:, 1], marker="o", linestyle="None", **plot_kwargs)
-
-    def plot(self, points, ax=None, **point_draw_kwargs):
-        """Plot points in the circle."""
+        _defaults = {"marker": "o", "linestyle": "None"}
+        self._graph_defaults["scatter"] = _defaults
+        self._graph_defaults["plot"] = _defaults
+
+    def plot_space(self, ax=None, color="black", **plot_kwargs):
```
Comment (ninamiolane) [inline geomstats/visualization/hypersphere.py:32]: Need docstring

File: geomstats/visualization/special_euclidean.py [hunk 6]
```diff
@@ -1,64 +1,20 @@
 """Visualization for Geometric Statistics."""
```
Comment (ninamiolane) [inline geomstats/visualization/special_euclidean.py:1]: Rename Visualization for special euclidean space

File: geomstats/visualization/hypersphere.py [hunk 7]
```diff
@@ -5,155 +5,100 @@
...
         ax = self.set_ax(ax=ax)
-        self.add_points(points)
-        self.draw(ax, **point_draw_kwargs)
+        ax.plot(self._circle_x, self._circle_y, color=color, **plot_kwargs)
+
+        return ax
 
 
-class Sphere:
+class Sphere(Plotter):
```
Comment (ninamiolane) [inline geomstats/visualization/hypersphere.py:39]: Add method `plot_points`

File: geomstats/visualization/special_euclidean.py [hunk 8]
```diff
@@ -1,64 +1,20 @@
...
+class SpecialEuclidean2(Plotter):
+    """Create the points point_a and point_b to plot a special euclidean space.
 
+    Create point_a and point_b to create the wire frame of a special euclidean space.
+    Their point_type is matrix.
+    """
 
-class SpecialEuclidean2:
-    """Class used to plot points in the 2d special euclidean group."""
+    def __init__(self, point_type="matrix"):
```
Comment (ninamiolane) [inline geomstats/visualization/special_euclidean.py:14]: missing plot_space
missing plot_points 
--> keep these two

**Review by ninamiolane — CHANGES_REQUESTED** [review]
@luisfpereira could you help me review this? Thnks!

File: geomstats/visualization/_plotting.py [hunk 9]
```diff
@@ -2,7 +2,7 @@
 
 import matplotlib.pyplot as plt
 
-import geomstats.backend as gs
+import geomstats._backend as gs
```
Comment (luisfpereira) [inline geomstats/visualization/_plotting.py:5]: `geomstats.backend` is in fact the correct import.

File: geomstats/visualization/special_euclidean.py [hunk 10]
```diff
@@ -1,64 +1,20 @@
...
-            translation[:, 0],
-            translation[:, 1],
-            frame_2[:, 0],
-            frame_2[:, 1],
-            width=0.005,
-            color="r",
-        )
-        ax.scatter(translation[:, 0], translation[:, 1], s=16, **kwargs)
+        self._space = SpecialEuclidean(point_a=[..., 2], point_b=[..., 2])
+        self._metric = SpecialEuclidean(point_a=[..., 2], point_b=[..., 2])
```
Comment (luisfpereira) [inline geomstats/visualization/special_euclidean.py:19]: What's your purpose with these lines? They do not work.

**Review by luisfpereira — CHANGES_REQUESTED** [review]
@SamFeinstein, the plotter that you've created cannot be instantiated as you passed wrong arguments to `SpecialEuclidean`.

**Pull request closed without merging**